client/checkout/woopay/direct-checkout/woopay-direct-checkout.js

```diff
@@ -127,39 +157,32 @@ class WoopayDirectCheckout {
 	}
 
 	/**
-	 * Adds a click-event listener that redirects to the WooPay checkout page to the given elements.
-	 *
-	 * @param {*[]} elements The elements to add a click-event listener to.
-	 */
-	static redirectToWooPaySession( elements ) {
-		elements.forEach( ( element ) => {
-			element.addEventListener( 'click', async ( event ) => {
-				event.preventDefault();
-
-				const woopayRedirectUrl = await this.sendRedirectSessionDataToWooPay();
-				this.teardown();
-
-				window.location.href = woopayRedirectUrl;
-			} );
-		} );
-	}
-
-	/**
-	 * Adds a click-event listener that redirects to WooPay and lets WooPay handle the checkout flow
-	 * to the given elements.
+	 * Adds a click-event listener to the given elements that redirects to the WooPay checkout page.
 	 *
 	 * @param {*[]} elements The elements to add a click-event listener to.
+	 * @param {boolean} useCheckoutRedirect Whether to use the `checkout_redirect` flag to let WooPay handle the checkout flow.
 	 */
-	static redirectToWooPay( elements ) {
+	static redirectToWooPay( elements, useCheckoutRedirect ) {
 		elements.forEach( ( element ) => {
 			element.addEventListener( 'click', async ( event ) => {
 				event.preventDefault();
 
-				const woopayRedirectUrl = await this.sendRedirectSessionDataToWooPay();
-				this.teardown();
-
-				window.location.href =
-					woopayRedirectUrl + '&woopay_checkout_redirect=1';
+				try {
+					let woopayRedirectUrl = await this.resolveWooPayRedirectUrl();
+					if ( useCheckoutRedirect ) {
+						woopayRedirectUrl += '&checkout_redirect=1';
+					}
+
+					this.teardown();
+					// TODO: Add telemetry as to _how long_ it took to get to this step.
+					window.location.href = woopayRedirectUrl;
+				} catch ( error ) {
+					// TODO: Add telemetry as to _why_ we've short-circuited the WooPay checkout flow.
```

Comment:
Suggested the creation of an issue for these `TODO` comments in the Epic: https://github.com/Automattic/woopay/issues/2440#issuecomment-1949596649.